### PR TITLE
Add support for md5 and md5sha1 hashes

### PR DESCRIPTION
**Pull request by qmuntal**
This PR updates `cryptoHashToMD` to support `crypto.MD5` and `crypto.MD5SHA1`. Those types are still not used, but will be when implementing RSA.

I've submitted these changes in a separate PR because supporting `crypto.MD5SHA1` is quite complex due to it not being supported in OpenSSL 1.0.2. When linking against OpenSSL 1.0.2 version we will provide a custom implementation backported from OpenSSL 1.1.0.

The custom implementation has been taken from RedHat master branch, as MSFT does not support `crypto.MD5SHA1` when using OpenSSL 1.0.2. We (MSFT) took this decision because backporting the code is more complex to implement without the OpenSSL headers, yet doable as demonstrated by this PR.

### CODE REVIEW

File: openssl/port_evp_md5_sha1.c
Comment (dagood) [inline openssl/port_evp_md5_sha1.c:68]: Maybe surround this section with a note? Looks like some header includes have been partially expanded?

File: openssl/port_evp_md5_sha1.c
Comment (dagood) [inline openssl/port_evp_md5_sha1.c:102]: Nitpick: Mixture of 2 and 4-space indents in the file.

**Review by derekparker — APPROVED** [review]
LGTM pending @ueno review.

File: openssl/port_evp_md5_sha1.c
Comment (qmuntal) [reply openssl/port_evp_md5_sha1.c:75]: Done!

File: openssl/port_evp_md5_sha1.c
Comment (qmuntal) [reply openssl/port_evp_md5_sha1.c]: Fixed!

**Comment by jaredpar** [comment]
@derekparker, @ueno 

This PR's getting pretty stale. At about ~3 weeks of inactivity here. Is there something we can do to move this along? Want to make sure we're not missing anything here that you all are expecting. 

Do understand if there are other time conflicts factoring in here.

**Comment by derekparker** [comment]
Sorry for the delay, just waiting on @ueno review. I will ping him to TAL at this.

**Review by ueno — APPROVED** [review]
Apologies for the delay. The changes looks good to me.

**Pull request merged**
